### PR TITLE
(maint) Drop ruby monkey patches

### PR DESCRIPTION
**Pull request by joshcooper**
Since puppet#main requires ruby 2.5 or greater, we can drop some patches for older ruby/openssl issues.

### CODE REVIEW

File: lib/puppet/util/monkey_patches.rb [hunk 1]
```diff
@@ -93,23 +86,4 @@ def set_default_paths
...
-                OpenSSL::X509::Name::RFC2253 & ~ASN1_STRFLGS_ESC_MSB
-              end
-
-      def to_utf8
-        # https://github.com/ruby/ruby/blob/v2_5_5/ext/openssl/ossl_x509name.c#L317
-        str = to_s(FLAGS)
-        str.force_encoding(Encoding::UTF_8)
-      end
-    end
-  end
```
Comment (GabrielNagy) [inline lib/puppet/util/monkey_patches.rb:114]: Looks like the JRuby openssl still doesn't implement `#to_utf8` for `OpenSSL::X509::Name`: https://github.com/jruby/jruby-openssl/blob/master/src/main/java/org/jruby/ext/openssl/X509Name.java

Comment (joshcooper) [reply lib/puppet/util/monkey_patches.rb:114]: :cry:

Comment (justinstoller) [reply lib/puppet/util/monkey_patches.rb:114]: Do we use this in JRuby though?

Comment (joshcooper) [reply lib/puppet/util/monkey_patches.rb]: Not really, but I try to run as much of the puppet code/tests on JRuby as possible.

Comment (joshcooper) [reply lib/puppet/util/monkey_patches.rb]: Ah so I forgot that things like ACE use openssl classes and they depend on puppet, so dropping the monkey patch *could* break them if they were to call `to_utf8`. I don't think they are, but to be on the safe side I think it's best to keep the monkey patch for JRuby.

Comment (justinstoller) [reply lib/puppet/util/monkey_patches.rb]: bummer, but good to know

**Review by GabrielNagy — APPROVED** [review]

**Pull request merged**